lib/dogapi/v1/aws_integration.rb

```diff
@@ -0,0 +1,113 @@
+      #   :role_name => '<AWS_ROLE_NAME>'
+      # }
+      #
+      # Access Key/Secret Access Key based accounts (GovCloud and China only)
+      #
+      # config = {
+      #   :access_key_id => '<AWS_ACCESS_KEY_ID>',
+      #   :host_tags => ['api:example'],
+      #   :secret_access_key => '<AWS_SECRET_ACCESS_KEY>'
+      # }
+      #
+      # dog = Dogapi::Client.new(api_key, app_key)
+      #
+      # puts dog.create_aws_integration(config)
+      def create_aws_integration(config)
+        request(Net::HTTP::Post, "/api/#{API_VERSION}/integration/aws", nil, config, true)
+      end
+
+      # Delete an integration
+      # :config => Hash: integration config.
+      # config = {
+      #   :account_id => '<AWS_ACCOUNT>',
+      #   :role_name => '<AWS_ROLE_NAME>'
+      # }
+      # Access Key/Secret Access Key based accounts (GovCloud and China only)
+      #
+      # config = {
+      #   :access_key_id => '<AWS_ACCESS_KEY_ID>',
+      # }
+      #
+      # dog = Dogapi::Client.new(api_key, app_key)
+      #
+      # puts dog.delete_aws_integration(config)
+      def delete_aws_integration(config)
+        request(Net::HTTP::Delete, "/api/#{API_VERSION}/integration/aws", nil, config, true)
+      end
+
+      # List available AWS namespaces
+      def list_aws_namespaces
+        request(Net::HTTP::Get, "/api/#{API_VERSION}/integration/aws/available_namespace_rules", nil, nil, false)
+      end
+
+      # Generate new AWS external ID for a specific integrated account
+      # :config => Hash: integration config.
+      # config = {
+      #   :account_id => '<AWS_ACCOUNT>',
+      #   :role_name => '<AWS_ROLE_NAME>'
+      # }
+      #
+      # dog = Dogapi::Client.new(api_key, app_key)
+      #
+      # puts dog.generate_external_id(config)
+      def generate_external_id(config)
+        request(Net::HTTP::Put, "/api/#{API_VERSION}/integration/aws/generate_new_external_id", nil, config, true)
+      end
+
+      # Update integrated AWS account.
+      # :config => Hash: integration config.
+      # existing_config = {
+      #   "account_id": '<EXISTING_AWS_ACCOUNT>',
+      #   "role_name": '<EXISTING_AWS_ROLE_NAME>'
+      # }
+      #
+      # config = {
+      #   "account_id": '<NEW_AWS_ACCOUNT>',
+      #   "host_tags": ['tag:example1,tag:example2'],
+      #   "filter_tags": ['datadog:true']
+      # }
+      #
+      # Access Key/Secret Access Key based accounts (GovCloud and China only)
+      #
+      # existing_config = {
+      #   "access_key_id": '<EXISTING_ACCESS_KEY_ID>',
+      #   "secret_access_key": '<EXISTING_SECRET_ACCESS_KEY>'
+      # }
+      #
+      # config = {
+      #   "access_key_id": '<NEW_ACCESS_KEY_ID>',
+      #   "host_tags": ['new:tags'],
+      #   "filter_tags": ['datadog:true']
+      # }
+      #
+      # dog = Dogapi::Client.new(api_key, app_key)
+
+      # puts dog.update_aws_account(existing_config, config)
+      def update_aws_account(existing_config, config)
```

Comment:
Why isn't this logic here as well? https://github.com/DataDog/datadogpy/pull/429/files#diff-ddb69aa9c795db6e6ba780d781222fc0R172-R186

Reply:
That logic was added in order to make it more user friendly. Based off of the feedback comment here: https://github.com/DataDog/datadogpy/pull/429#discussion_r334398238

I've now modified this so that it's more in line with that train of thought. i.e. it now accepts `config` and `new_config` where it had been `existing_config` and `config`.